### PR TITLE
Feature/kafka timeout handling

### PR DESCRIPTION
**Pull request by BEagle1984**
@msallin would you mind having a quick look at this PR? I'm trying some things to improve the situation with the consumers getting stuck.
I'm unsure whether I solved the actual issue in the retry mechanism, since I haven't been able to reproduce it so far, but I'm also trying to recover from the poll timeout error that is eventually thrown. It's probably not the most elegant implementation ever, but it does what it's meant to do...in my tests, as always, then we will see how it behaves on prod.

### CODE REVIEW

File: src/Silverback.Integration/Messaging/Broker/Consumer.cs [hunk 1]
```diff
@@ -210,9 +228,23 @@ public async Task TriggerReconnectAsync()
...
-                        await WaitUntilConsumingStoppedAsync().ConfigureAwait(false);
-                        await DisconnectAsync().ConfigureAwait(false);
-                        await ConnectAsync().ConfigureAwait(false);
+                        try
+                        {
+                            await WaitUntilConsumingStoppedAsync().ConfigureAwait(false);
+                            await DisconnectAsync().ConfigureAwait(false);
+                            await ConnectAsync().ConfigureAwait(false);
+
+                            _isReconnecting = false;
```
Comment (msallin) [inline src/Silverback.Integration/Messaging/Broker/Consumer.cs:237]: Log the successful reconnect?

File: src/Silverback.Integration/Messaging/Broker/Consumer.cs [hunk 2]
```diff
@@ -210,9 +228,23 @@ public async Task TriggerReconnectAsync()
             Task.Run(
```
Comment (msallin) [inline src/Silverback.Integration/Messaging/Broker/Consumer.cs:228]: why starting this in a new task?

File: src/Silverback.Integration.Kafka/Messaging/Broker/KafkaConsumer.cs [hunk 3]
```diff
@@ -379,6 +380,8 @@ private void InitConfluentConsumer()
 
                 Endpoint.TopicPartitions.ForEach(
```
Comment (msallin) [inline src/Silverback.Integration.Kafka/Messaging/Broker/KafkaConsumer.cs:381]: If I unterstand correctly, if subscribing the channel manager is set the partitions are assigned.
Wouldn't it be more elegant to also use the same mechanism here and do not distinguierst these cases?

**Review by msallin — APPROVED** [review]
Somethings I noticed while looking at the code:
- In ConsumeOnce, maybe worth to add a logging entry when the while takes long (currently when it hangs here, we see only the low level trace)?
- Be aware of preventable allocations in hot paths which depend on log level
e.g. ConsumerChannelsManager.Write LogConsumerLowLevelTrace always allocates a new delegate, a new object and 3 variables are boxed. This is also the case if the log will never be written.
- Maybe its worth some time to look at closure and delegate allocations in hot paths some day: https://devblogs.microsoft.com/pfxteam/know-thine-implicit-allocations/, there are a bunch of them.
- KafkaConsumer.InitAndStartConsumeLoopHandler: Is LoC 392 & 393 is necessary?
- In KafkaConsumer.InitAndStartConsumeLoopHandler, if the IsConsuming flag is false, no consumation will be started and nobody will ever notice. It was a bit hard to follow the control flow with my knowledge. Maybe it isn't possible.
However, is it worth to be logged? Or even throw an exception if its not an expected state?

File: src/Silverback.Integration.HealthChecks/Messaging/HealthChecks/HealthCheckBuilderExtensions.cs [hunk 4]
```diff
@@ -119,22 +122,54 @@ public static class HealthCheckBuilderExtensions
...
             this IHealthChecksBuilder builder,
             string name = "Consumers",
-            HealthStatus? failureStatus = default,
-            IEnumerable<string>? tags = default)
+            HealthStatus? failureStatus = null,
+            IEnumerable<string>? tags = null) =>
+            AddConsumersCheck(builder, ConsumerStatus.Ready, name, failureStatus, tags);
+
+        /// <summary>
+        ///     Adds an health check that verifies that all consumers are connected.
```
[the remark this answers is not in the document]
Comment (BEagle1984) [reply src/Silverback.Integration.HealthChecks/Messaging/HealthChecks/HealthCheckBuilderExtensions.cs:130]: I just fixed it in the docs... 🤣

File: src/Silverback.Integration.HealthChecks/Messaging/HealthChecks/HealthCheckBuilderExtensions.cs [hunk 5]
```diff
@@ -119,22 +122,54 @@ public static class HealthCheckBuilderExtensions
...
+        /// </param>
+        /// <param name="name">
+        ///     The health check name. If <c>null</c> the name 'OutboundQueue' will be used for the name.
+        /// </param>
+        /// <param name="failureStatus">
+        ///     The <see cref="HealthStatus" /> that should be reported when the health check fails. Optional. If
+        ///     <c>null</c> then the default status of <see cref="HealthStatus.Unhealthy" /> will be reported.
+        /// </param>
+        /// <param name="tags">
+        ///     A list of tags that can be used to filter sets of health checks. Optional.
```
[the remark this answers is not in the document]
Comment (BEagle1984) [reply src/Silverback.Integration.HealthChecks/Messaging/HealthChecks/HealthCheckBuilderExtensions.cs:146]: I think I copied this from msft, not sure though. I will check.

File: src/Silverback.Integration/Messaging/Broker/Consumer.cs (hunk 1 above)
Comment (BEagle1984) [reply src/Silverback.Integration/Messaging/Broker/Consumer.cs:237]: The `ConnectAsync` will log an INFO already. 👍

**Comment by BEagle1984** [comment]
I will manually merge it. 👍
